frontend/packages/dev-console/src/components/pipelines/pipeline-form/SecretForm.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikValues, useFormikContext } from 'formik';
+import { ActionGroup, Button, ButtonVariant } from '@patternfly/react-core';
+import { CheckIcon, CloseIcon } from '@patternfly/react-icons';
+import { Dropdown, ButtonBar } from '@console/internal/components/utils';
+import {
+  SecretType,
+  BasicAuthSubform,
+  SSHAuthSubform,
+  CreateConfigSubform,
+} from '@console/internal/components/secrets/create-secret';
+import './SecretForm.scss';
+
+const authTypes = {
+  [SecretType.dockerconfigjson]: 'Image Registry Credentials',
+  [SecretType.basicAuth]: 'Basic Authentication',
+  [SecretType.sshAuth]: 'SSH Key',
+};
+
+const renderSecretForm = (type, stringData, onDataChanged) => {
+  switch (type) {
+    case SecretType.basicAuth:
+      return <BasicAuthSubform onChange={onDataChanged} stringData={stringData} />;
+    case SecretType.sshAuth:
+      return <SSHAuthSubform onChange={onDataChanged} stringData={stringData} />;
+    case SecretType.dockerconfigjson:
+      return <CreateConfigSubform onChange={onDataChanged} stringData={stringData} />;
+    default:
+      return null;
+  }
+};
+
+const SecretForm: React.FC<FormikValues> = ({
+  handleSubmit,
+  handleReset,
+  status,
+  isSubmitting,
+}) => {
+  const [name, setName] = React.useState('');
+  const [authType, setAuthType] = React.useState(SecretType.dockerconfigjson);
+  const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const stringData = {};
+  const onNameChanged = (event) => {
+    setName(event.target.value);
+    setFieldValue('secretName', event.target.value);
+  };
```

Comment:
Uh... what? Set the name locally, and set the name in the form? Why not just set it in the form and use that instead of holding onto a local state of the name?